### PR TITLE
[TT-11011] Add OAS operations.Internal

### PR DESCRIPTION
**Pull request by titpetric**
Implements internal access to endpoints in OAS.

Parent: https://tyktech.atlassian.net/browse/TT-4859 
Subtask: https://tyktech.atlassian.net/browse/TT-11011

### CODE REVIEW

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5946/merge
Commit: b9c8991d5a1a2938d9b1277b8f1f7d8bec9a909b 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7553999268)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5946/merge
Commit: b9c8991d5a1a2938d9b1277b8f1f7d8bec9a909b 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7553999268)

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/pull/5946/merge
Commit: b9c8991d5a1a2938d9b1277b8f1f7d8bec9a909b 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7553999268)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/5946/merge
Commit: b9c8991d5a1a2938d9b1277b8f1f7d8bec9a909b 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7553999268)

File: apidef/oas/oas_test.go [hunk 1]
```diff
@@ -215,8 +215,6 @@ func TestOAS_ExtractTo_ResetAPIDefinition(t *testing.T) {
 		"APIDefinition.VersionData.Versions[0].ExtendedPaths.TrackEndpoints[0].Method",
 		"APIDefinition.VersionData.Versions[0].ExtendedPaths.DoNotTrackEndpoints[0].Path",
 		"APIDefinition.VersionData.Versions[0].ExtendedPaths.DoNotTrackEndpoints[0].Method",
-		"APIDefinition.VersionData.Versions[0].ExtendedPaths.Internal[0].Path",
```
Comment (andrei-tyk) [inline apidef/oas/oas_test.go:218]: can you please provide some context as of why are those lines removed? It's a bit unclear for me

Comment (titpetric) [reply apidef/oas/oas_test.go:218]: > 	// The expectedFields value lists fields that do not support migration.
	// When adding a migration for ExtendedPaths sections, clear the list of
	// fields below, and clear the value in ExtendedPaths.Clear() function.

assert.Equal(t, expectedFields, noOASSupportFields)

Once a field is extracted from OAS, it should be cleared in apidef, so as to not keep any previous values.

**Review by andrei-tyk — APPROVED** [review]

File: apidef/oas/internal.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package oas
+
+import "github.com/TykTechnologies/tyk/apidef"
+
+// Internal holds the endpoint configuration, configuring the endpoint for internal requests.
+// Tyk classic API definition: `version_data.versions[].extended_paths.internal`.
```
Comment (titpetric) [inline apidef/oas/internal.go:6]:
```suggestion
// Tyk classic API definition: `version_data.versions..extended_paths.internal[*]`.
```
